torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -4562,6 +4563,67 @@ def test_ddp_hook_with_optimizer_parity_sgd(self, optimize_subset):
                 weight_decay=sgd_weight_decay,
             )
 
+        def _test_ddp_apply_optim_in_backward(
+            self, optim_cls, optim_kwargs, register_hook=True,
```

Comment:
Why make `register_hook` a default argument?

We could just not default it, and then explicitly pass `True` or `False` e.g. as part of a subtest.